### PR TITLE
TT-2721: tests for unused optional variables

### PR DESCRIPTION
**Pull request by urbanishimwe**
for more info check https://tyktech.atlassian.net/browse/TT-2721

### CODE REVIEW

File: pkg/graphql/normalization_test.go [hunk 1]
```diff
@@ -95,7 +95,41 @@ func TestRequest_Normalize(t *testing.T) {
...
 
-	t.Run("should successfully normalize query and remove unused variables with no value provided", func(t *testing.T) {
+	t.Run("should successfully normalize query and remove unused variables and values", func(t *testing.T) {
+		const expectedVar = "query MySearch($s: String!){\n    search(name: $s){\n        ... on Human {\n            name\n        }\n    }\n}"
+		for _, v := range []Request{
+			{
+				OperationName: "MySearch",
+				Variables: stringify(map[string]interface{}{
+					"s":  "Luke",
+					"s2": nil,
```
Comment (jensneuse) [inline pkg/graphql/normalization_test.go:105]: Are you sure that this will marshal to "s2": null in the final JSON?

Comment (urbanishimwe) [reply pkg/graphql/normalization_test.go:105]: @jensneuse  I'm not sure, just a test case I wanted to cover! do you have a better idea?

**Review by pvormste — APPROVED** [review]
Seems that tests are covering the issue. LGTM

File: pkg/graphql/normalization_test.go (hunk 1 above)
Comment (jensneuse) [reply pkg/graphql/normalization_test.go:105]: Instead of stringify, you could create another test where you hardcode "s2":null into the JSON string/bytes to make sure this case is covered.

Comment (urbanishimwe) [reply pkg/graphql/normalization_test.go:105]: @jensneuse done https://github.com/TykTechnologies/graphql-go-tools/pull/141/commits/c208e7e7ed4fb1766a69157fc68e4933f31d5e40
